File: connectors/connector-trello/src/main/scala/com/taskadapter/connector/trello/TrelloFactory.scala
package com.taskadapter.connector.trello

import com.google.gson.{JsonElement, JsonParseException}
import com.taskadapter.connector.common.ConfigUtils
import com.taskadapter.connector.definition.{Descriptor, PluginFactory, WebConnectorSetup}
import com.taskadapter.model.Field

class TrelloFactory extends PluginFactory[TrelloConfig, WebConnectorSetup] {

  override def getDescriptor = new Descriptor(TrelloConnector.ID, "Trello")

  override def getAllFields: java.util.List[Field[_]] = TrelloField.fields

  override def createConnector(config: TrelloConfig, setup: WebConnectorSetup) = new TrelloConnector(config, setup)

  override def writeConfig(config: TrelloConfig): JsonElement = ConfigUtils.createDefaultGson.toJsonTree(config)

  @throws[JsonParseException]
  override def readConfig(config: JsonElement): TrelloConfig = ConfigUtils.createDefaultGson.fromJson(config, classOf[TrelloConfig])

  override def createDefaultConfig = new TrelloConfig()

  override def getDefaultFieldsForNewConfig: java.util.List[Field[_]] = TrelloField.defaultFieldsForNewConfig
}

File: connectors/connector-trello/src/main/java/com/taskadapter/connector/trello/TrelloFactory.java
package com.taskadapter.connector.trello;

import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.taskadapter.connector.NewConnector;
import com.taskadapter.connector.common.ConfigUtils;
import com.taskadapter.connector.definition.Descriptor;
import com.taskadapter.connector.definition.PluginFactory;
import com.taskadapter.connector.definition.WebConnectorSetup;
import com.taskadapter.model.Field;

import java.util.List;

public class TrelloFactory implements PluginFactory<TrelloConfig, WebConnectorSetup> {

    @Override
    public List<Field<?>> getAllFields() {
        return TrelloField.fields;
    }

    @Override
    public List<Field<?>> getDefaultFieldsForNewConfig() {
        return TrelloField.defaultFieldsForNewConfig();
    }

    @Override
    public NewConnector createConnector(TrelloConfig config, WebConnectorSetup setup) {
        return new TrelloConnector(config, setup);
    }

    @Override
    public Descriptor getDescriptor() {
        return new Descriptor(TrelloConnector.ID, "Trello");
    }

    @Override
    public JsonElement writeConfig(TrelloConfig config) {
        return ConfigUtils.createDefaultGson().toJsonTree(config);
    }

    @Override
    public TrelloConfig readConfig(JsonElement config) throws JsonParseException {
        return ConfigUtils.createDefaultGson()
                .fromJson(config, TrelloConfig.class);
    }

    @Override
    public TrelloConfig createDefaultConfig() {
        return new TrelloConfig();
    }
}
